sympy/solvers/tests/test_solvers.py

```diff
@@ -1310,6 +1310,12 @@ def test_checksol():
     assert checksol([x - 1, x**2 - 1], x, 1) is True
     assert checksol([x - 1, x**2 - 2], x, 1) is False
     assert checksol(Poly(x**2 - 1), x, 1) is True
+    assert checksol(0, {}) is True
+    assert checksol([1e-10, x - 2], x, 2) is False
```

Comment:
Since `numerical=True` is the default, why is this giving False?

Reply:
If you're talking about the last assertion (`assert checksol([1e-10, x - 2], x, 2) is False`), I think we agreed (https://github.com/sympy/sympy/pull/22866#discussion_r802574214 and https://github.com/sympy/sympy/pull/22866#discussion_r802618221) that for equations that are just plain numbers, `checksol` should only return `True` if the equation is exactly zero. In this case the `1e-10` "equation" is not zero, so `checksol` returns `False`. Did I misunderstand something?

Reply:
OK, that makes sense: no tolerance for a number that is already not zero. I don't think the case of no free symbols needs special handling -- that's one of those corner cases that I really don't envision this dealing with. So returning None instead of False or True seems ok.